lib/tasks/test.rake

```diff
@@ -16,4 +16,36 @@ namespace :test do
     t.verbose = true
     t.warning = false
   end
+
+  namespace :syntax do
+    desc "Validate selected template snapshots against ksvalidator with EL7 profile"
+    task :kickstart_el7 do
+      Dir.glob('test/unit/foreman/renderer/snapshots/ProvisioningTemplate/provision/*Kickstart*').each do |file|
+        puts "Validating #{file}"
+        `ksvalidator --version RHEL7 "#{file}"`
+        abort("invalid kickstart: #{file}") unless $?.success?
+      end
+    end
+
+    desc "Validate selected template snapshots against ksvalidator with EL8 profile"
+    task :kickstart_el8 do
+      Dir.glob('test/unit/foreman/renderer/snapshots/ProvisioningTemplate/provision/*Kickstart*').each do |file|
+        puts "Validating #{file}"
+        `ksvalidator --version RHEL8 "#{file}"`
+        abort("invalid kickstart: #{file}") unless $?.success?
+      end
+    end
+
+    desc "Validate selected template snapshots against ksvalidator with EL8 profile"
+    task :grub2 do
+      Dir.glob('test/unit/foreman/renderer/snapshots/ProvisioningTemplate/PXEGrub2/*').each do |file|
+        puts "Validating #{file}"
+        `grub2-script-check "#{file}"`
+        abort("invalid grub config: #{file}") unless $?.success?
+      end
+    end
+
+    desc "Validate selected template snapshots"
+    task :all => [:kickstart_el7, :grub2]
```

Comment:
What about kickstart_el8? Why don't we have a `kickstart` task that goes over all supported RHEL versions? I'm sure we'll have a RHEL9 at some point in the future.

Reply:
As I said in the description, it currently fails due to deprecation which is already reported as an error by the tool. I don't want to sove this in this PR, thus I am creating the task but we would not call this from jenkins for now. This is security related and it will involve some more testing.